tools/accuracy_checker/accuracy_checker/postprocessor/resize.py

```diff
@@ -0,0 +1,172 @@
+try:
+    from PIL import Image
+except ImportError:
+    Image = None
+
+
+class Resize(PostprocessorWithSpecificTargets):
+
+    __provider__ = 'resize'
+
+    prediction_types = (DepthEstimationPrediction, StyleTransferPrediction,
+                        SegmentationPrediction, SuperResolutionPrediction, )
+    annotation_types = (DepthEstimationAnnotation, StyleTransferAnnotation,
+                        SegmentationAnnotation, SuperResolutionAnnotation, )
+
+    @classmethod
+    def parameters(cls):
+        parameters = super().parameters()
+        parameters.update({
+            'dst_width': NumberField(
+                value_type=int, optional=True, min_value=1, description="Destination width for resize"
+            ),
+            'dst_height': NumberField(
+                value_type=int, optional=True, min_value=1, description="Destination height for resize."
+            ),
+            'size': NumberField(
+                value_type=int, optional=True, min_value=1,
+                description="Destination size for resize for both dimensions (height and width)."
+            )
+        })
+        return parameters
+
+    def configure(self):
+        if Image is None:
+            raise ValueError('{} requires pillow, please install it'.format(self.__provider__))
+        self.dst_height, self.dst_width = get_size_from_config(self.config, allow_none=True)
+
+    def process_image(self, annotations, predictions):
+        if self.dst_width and self.dst_height:
+            height = self.dst_height
+            width = self.dst_width
+        else:
+            height = self.image_size[0]
+            width = self.image_size[1]
+
+        @singledispatch
+        def resize(entry, height, width):
+            return entry
+
+        @resize.register(DepthEstimationPrediction)
+        def _(entry, height, width):
+            entry.depth_map = cv2.resize(entry.depth_map, (width, height))
+
+            return entry
+
+        @resize.register(StyleTransferAnnotation)
+        def _(entry, height, width):
+            data = Image.fromarray(entry.value)
+            data = data.resize((width, height), Image.BICUBIC)
+            entry.value = np.array(data)
+
+            return entry
+
+        @resize.register(SegmentationPrediction)
+        def _(entry, height, width):
+            if len(entry.mask.shape) == 2:
+                entry.mask = self._segm_resize(entry.mask, width, height)
+                return entry
+
+            entry_mask = []
+            for class_mask in entry.mask:
+                resized_mask = self._segm_resize(class_mask, width, height)
+                entry_mask.append(resized_mask)
+            entry.mask = np.array(entry_mask)
+
+            return entry
+
+        @resize.register(SegmentationAnnotation)
+        def _(entry, height, width):
+            entry.mask = self._segm_resize(entry.mask, width, height)
+            return entry
+
+        @resize.register(SuperResolutionPrediction)
```

Comment:
my opinion that resize should be applied for annotation and prediction in all cases if we provide opprotunity to select target. It looks suspicious if user specify dst size and apply_to annotation and it do nothing